source/s3_auto_ranged_put.c

```diff
@@ -122,6 +151,11 @@ static int s_parse_resume_info_from_token(
     aws_json_module_cleanup();
 
     return AWS_ERROR_SUCCESS;
+
+invalid_argument_cleanup:
+    aws_json_value_destroy(root);
+    aws_json_module_cleanup();
+    return AWS_ERROR_INVALID_ARGUMENT;
```

Comment:
just `return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);`

We usually just return AWS_OP_ERROR/AWS_OP_SUCCESS, the detailed error code will be raised within the function.